### PR TITLE
Prepare 9.3.3

### PR DESCRIPTION
**Pull request by bourtemb**
Update cppTango to release 9.3.3
Update TangoAccessControl to release 2.14
Update assets/README to show changes between 9.2.5a and 9.3.3
Update assets/TANGO_CHANGES for 9.3.3 with a minimal text explanation

### CODE REVIEW

**Comment by Ingvord** [comment]
Should we merge #16 first?

**Comment by Ingvord** [comment]
I would change rest-server to 1.9 as well

**Comment by bourtemb** [comment]
Good question... I missed that one...

**Comment by Ingvord** [comment]
There is a bunch of changes in the README. Maybe we should extract them to assets/CHANGES_LOG file or merge it with TANGO_CHANGES?

**Comment by bourtemb** [comment]
> There is a bunch of changes in the README. Maybe we should extract them to assets/CHANGES_LOG file or merge it with TANGO_CHANGES?

I think there is a big work to be done to harmonize all these files and to make them up to date.

Some sections in the README file are overlapping what is/should be in other files.
I think all the installation instructions should be in the INSTALL file (which could be upgraded to INSTALL.md).
All the Changes should be in a separate file too.
It looks like there are different kinds of changes which were listed before: the release versions of the different SW parts and the bugs and changes in C++ Tango library.
There was not much information about the bugs and changes in other SW (Astor, JTango,...).

The questions are:
Do we change that?
How do we organize that?
What kind of information do we want to provide to the user?

I think there is a big work to be done to ensure the content of these files is up to date.

I guess it should not be too complicated to generate automatically a file from assets/distribution.properties which would be listing all the different versions used for the different provided Tango SW components.

About the Changelog, I am not sure about the level of details we would provide here...
If we provide some details, do we do it for the changes in all SW components, not only cppTango?

I guess it would be good to provide RELEASE NOTES too.
Again, we need to agree on the level of details we provide.

**Comment by Ingvord** [comment]
From my experience in such aggregated distribution the best is to refer to every particular repo aka README, CHANGE_LOG, RELEASE_NOTES etc individually via a link. 

While here to provide an info on migration e.g. what needs to be done to ensure 9.2.5a -> 9.3.3 migration (possibly not much), but if have such an info for each Tango release it is much more convenient for users to track what needs to be done to migrate from Tango 7.1.2 (arbitrary number) to 9.3.3

I agree a discussion would be helpful

**Comment by mliszcz** [comment]
Hi @bourtemb, @Ingvord 

CC @piogor

What is the current status of 9.3.3? Have any critical issues been found?

There were no issues reported in kernel in last month (at least there's nothing new on github).
In source distribution we have #24 and #25 - root cause is known and possible corrections were proposed.
What about #21, #22 and #23 - do you plan these enhancements for 9.3.3?

We plan to include Tango 9.3.3 in Tangobox 9.3. I'm just wondering whether we could use this branch (`prepare-9.3.3`) and patch #24/#25 manually or should we wait for an official release.

**Comment by bourtemb** [comment]
Hi @mliszcz,

I'm just coming back from holidays. I didn't have time yet to comment the PRs or issues you created.

No big issue were reported indeed.
I think there is still quite a lot of work to be done on the Tango distribution to release it.
We should definitely integrate #24 and #25.
I think we should also solve #21 and #23 at least.
#22 would be great too because it would avoid to generate a liblog4tango library which should no longer exist.
There is a big work to be done to update and cleanup files like the assets/README assets/TANGO_CHANGES and assets/INSTALL files.

**Comment by Ingvord** [comment]
Hi, @mliszcz , @bourtemb ,

Here is my 50 cents:

#24, #25, #21 - must be included in 9.3.3 before release.

#23 I am not sure, maybe if it won't break anything should be added as well.

#22 will require considerable amount of manpower (as per my experience with #8) so probably will be done in the next release,  as well as other "big work" on assets.

**Comment by mliszcz** [comment]
Hi @bourtemb, @Ingvord 

Thank you guys for your inputs. I'll submit a PR for #24 and #25 still today.
If no one will take care for #21, I may try to take a look next week - depends whether we will have spare cycles at S2Innovation.

PS. Reynald, I hope you had a great holiday!

File: distribution.properties [hunk 1]
```diff
@@ -22,7 +22,7 @@ dbbench-ver=1.3
 jssh-terminal-ver=1.13
 log-viewer-ver=2.0.5
 pogo-ver=9.6.18
-rest-server-ver=1.9
+rest-server-ver=1.11
```
Comment (bourtemb) [inline distribution.properties]: @Ingvord , could you please also update the README?
https://github.com/tango-controls/TangoSourceDistribution/blob/prepare-9.3.3/assets/README

Comment (Ingvord) [reply distribution.properties]: Done in f409a5a

**Comment by picca** [comment]
Hello could you re-add the source of the documentation, in order to be able to re-generate the documention during the build process on the Debain bbuild farm.

thanks

**Comment by picca** [comment]
Is it possible to provide a migration script of the database from n-1 to n, instead of  updating all migrations scripts at each release, thanks

Frederic

**Comment by picca** [comment]
Here a faillure during the build

https://salsa.debian.org/science-team/tango/-/jobs/185931

**Comment by picca** [comment]
```
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../lib/cpp/client -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/server -I/usr/include -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wl,-z,relro -I/usr/include  -D_TANGO_LIB -g -O2 -fdebug-prefix-map=/tmp/building/package=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o zmqeventsupplier.lo zmqeventsupplier.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../lib/cpp/client -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/log4tango/include -I../../../lib/cpp/server -I/usr/include -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wl,-z,relro -I/usr/include -D_TANGO_LIB -g -O2 -fdebug-prefix-map=/tmp/building/package=. -fstack-protector-strong -Wformat -Werror=format-security -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c zmqeventsupplier.cpp  -fPIC -DPIC -o .libs/zmqeventsupplier.o
zmqeventsupplier.cpp: In member function 'void Tango::ZmqEventSupplier::create_mcast_socket(std::__cxx11::string&, int, Tango::ZmqEventSupplier::McastSocketPub&)':
zmqeventsupplier.cpp:689:18: error: cannot convert 'zmq::socket_t' to 'void*'
     if (zmq_bind(*(ms.pub_socket),ms.endpoint.c_str()) != 0)
                  ^~~~~~~~~~~~~~~~
In file included from /usr/include/zmq.hpp:38,
                 from ../../../lib/cpp/client/event.h:41,
                 from ./tango.h:146,
                 from zmqeventsupplier.cpp:33:
/usr/include/zmq.h:482:32: note:   initializing argument 1 of 'int zmq_bind(void*, const char*)'
 ZMQ_EXPORT int zmq_bind (void *s_, const char *addr_);
                          ~~~~~~^~
make[6]: *** [Makefile:785: zmqeventsupplier.lo] Error 1
```

**Comment by Ingvord** [comment]
@picca , thanks for the info and the efforts to build Debian package out of this branch. This is already addressed in [cppTango-530](https://github.com/tango-controls/cppTango/pull/530). I guess we will have to wait for a merge...

**Comment by bourtemb** [comment]
> Is it possible to provide a migration script of the database from n-1 to n, instead of updating all migrations scripts at each release, thanks

I think we should still provide scripts for users migrating from Tango 7 and Tango 8.
But it's true that these scripts could also call the other scripts...
For instance, if a user wants to upgrade from Tango 7, the Tango 7 migration script would contain the DB queries to update from Tango 7 to Tango 8, and then simply call the Tango 8 migration script.
This would avoid code duplication.
Still, this means that we would need to update all the scripts at each source distribution release implying a DB change.

Why do you consider the update of all these scripts as a problem?

**Comment by picca** [comment]
Hello,

this is juste that dbconfig-common does an incremental upgrade from on version to the other.
So I just need a script whcih does the migration from 9.2.5 to 9.3.3

DELETE FROM property_class WHERE class='DServer' AND count >= 11;
INSERT INTO property_class VALUES('DServer','AllowedAccessCmd',11,'EventConfirmSubscription',NOW(),NOW(),NULL);


Is it the only change  ?

**Comment by bourtemb** [comment]
> Is it the only change ?

Yes, it is.

**Comment by bourtemb** [comment]
Actually, I think there is also something related to the rest server:

```
#
# Create entry for TangoRestServer device server in device table
#

DELETE FROM device WHERE server='TangoRestServer/rest';

INSERT INTO device VALUES ('sys/rest/0',NULL,'sys','rest','0',0,'nada','nada','TangoRestServer/rest',0,'TangoRestServer','nada',NULL,NULL,'nada');
INSERT INTO device VALUES ('dserver/TangoRestServer/rest',NULL,'dserver','TangoRestServer','rest',0,'nada','nada','TangoRestServer/rest',0,'DServer','nada',NULL,NULL,'nada');
```

But maybe this is not yet in the update scripts.

**Pull request merged**